src/api/crypto/Hash.cpp

```diff
@@ -361,41 +287,59 @@ auto Hash::Scrypt(
     return scrypt_.Generate(input, salt, N, r, p, bytes, writer);
 }
 
-auto Hash::sha_256_double(
-    const void* input,
-    const std::size_t size,
-    void* output) const noexcept -> bool
+auto Hash::sha_256_double(const ReadView data, const AllocateOutput destination)
+    const noexcept -> bool
 {
-    auto temp = space(Provider::HashSize(opentxs::crypto::HashType::Sha256));
+    try {
+        auto temp = Space{};
+        const auto rc =
+            Digest(opentxs::crypto::HashType::Sha256, data, writer(temp));
 
-    if (false ==
-        digest(opentxs::crypto::HashType::Sha256, input, size, temp.data())) {
-        LogError()(OT_PRETTY_CLASS())("Failed to calculate intermediate hash.")
-            .Flush();
+        if (false == rc) {
+
+            throw std::runtime_error{"failed to calculate intermediate hash"};
+        }
+
+        return Digest(
+            opentxs::crypto::HashType::Sha256, reader(temp), destination);
+    } catch (const std::exception& e) {
+        LogError()(OT_PRETTY_CLASS())(e.what()).Flush();
 
         return false;
     }
-
-    return digest(
-        opentxs::crypto::HashType::Sha256, temp.data(), temp.size(), output);
 }
 
 auto Hash::sha_256_double_checksum(
-    const void* input,
-    const std::size_t size,
-    void* output) const noexcept -> bool
+    const ReadView data,
+    const AllocateOutput destination) const noexcept -> bool
 {
-    auto temp = space(Provider::HashSize(opentxs::crypto::HashType::Sha256));
+    try {
+        auto temp = Space{};
+        const auto rc = sha_256_double(data, writer(temp));
```

Comment:
Where is `writer` taken from? It seems to not be either a function argument nor a member of `Hash` class. The same for `Hash::sha_256_double` and `Hash::bitcoin_hash_160`.

Reply:
writer (and Space) are declared in Bytes.hpp.